src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1012,8 +1006,9 @@ public void RemoveTextContainer(SourceTextContainer textContainer)
                         }
                         else
                         {
-                            _project._workspace.ApplyChangeToWorkspace(w => {
-                                w.OnDocumentClosed(documentId, new SourceTextLoader(textContainer, filePath: null));
+                            _project._workspace.ApplyChangeToWorkspace(w =>
+                            {
+                                // removing document automatically close the file if it is opened
```

Comment:
Does it? I see OnDocumentRemoved calling CheckDocumentCanBeRemoved which is checking open/closed state.

Reply:
interesting. what I looked was this (ClearDocumentData) which calls
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Workspace_Editor.cs,106 

that code makes sure it removes document from OpenFile tracking. which is only called from RemoveDoc/AdditionalDoc..

anyway, I will put it back